draft-polli-resource-digests-http.md

```diff
@@ -600,16 +638,45 @@ Response:
 
 # Security Considerations
 
+## Broken cryptographic algorithms
+
+Cryptographic algorithms are supposed to provide a proof of integrity
+usable eg. in signatures.
+
+To discourage mistakes, broken-cryptographic algorithms
+like `MD5` and `SHA-1` are now NOT RECOMMENDED.
+
+## Digest for end-to-end integrity
+
+`Digest` alone does not provide end-to-end integrity
+of HTTP messages over multiple hops, as it just covers
+the representation-data and not the representation metadata.
+
+Besides, it allows to protect representation data from
+buggy manipulation, buggy compression, etc.
+
+Moreover identity digest algorithms (eg. ID-SHA-256 and ID-SHA-512)
+allow piecing together a resource from different sources
+(e.g. different servers that perhaps apply different content codings)
+enabling the user-agent to detect that the application-layer tasks completed properly,
+before handing off to say the HTML parser, video player etc.
+
+Even a simple mechanism for end-to-end validation is thus valuable.
+
 ## Usage in signatures
 
 Digital signatures are widely used together with checksums to provide
 the certain identification of the origin of a message [NIST800-32].
 
-It's important to note that, being the Digest header an hash of a resource representation,
+It's important to note that, being the `Digest` header an hash of a resource representation,
 signing only the `Digest` header, without all the representation metatada (eg.
 the values of `Content-Type` and `Content-Encoding`) may expose the communication
 to tampering.
 
+`Digest` SHOULD always be used over a connection which provides
+integrity at transport layer that protects HTTP headers.
+
+A `Digest` header using NOT RECOMMENDED {digest-algorithms} MUST NOT be used in signatures.
```

Comment:
What does an endpoint do if it receives such a header. Reject the message and issue an error? This is usually written like "An endpoint that receives such a header SHOULD ..."

Reply:
maybe remove this statement or lowercasing the MUST NOT as a shouldn't?